spec.html

```diff
@@ -608,35 +685,53 @@ <h1>%TypedArray%.prototype.slice ( _start_, _end_ )</h1>
-            1. Let _srcByteOffset_ be _O_.[[ByteOffset]].
-            1. Let _targetByteIndex_ be _A_.[[ByteOffset]].
-            1. Let _srcByteIndex_ be (_k_ &times; _elementSize_) + _srcByteOffset_.
-            1. Let _limit_ be _targetByteIndex_ + _count_ &times; _elementSize_.
-            1. Repeat, while _targetByteIndex_ &lt; _limit_,
-              1. Let _value_ be GetValueFromBuffer(_srcBuffer_, _srcByteIndex_, ~Uint8~, *true*, ~Unordered~).
-              1. Perform SetValueInBuffer(_targetBuffer_, _targetByteIndex_, ~Uint8~, _value_, *true*, ~Unordered~).
-              1. Set _srcByteIndex_ to _srcByteIndex_ + 1.
-              1. Set _targetByteIndex_ to _targetByteIndex_ + 1.
+          1. <ins>Let _srcBuffer_ be _O_.[[ViewedArrayBuffer]].</ins>
+          1. <ins>Let _srcBufferByteLength_ be _regetBufferByteLength_(_srcBuffer_).</ins>
+          1. <del>If _srcType_ is different from _targetType_, then</del>
+            1. <del>Let _n_ be 0.</del>
+            1. <del>Repeat, while _k_ &lt; _final_,</del>
+              1. <del>Let _Pk_ be ! ToString(𝔽(_k_)).</del>
+              1. <del>Let _kValue_ be ! Get(_O_, _Pk_).</del>
+              1. <del>Perform ! Set(_A_, ! ToString(𝔽(_n_)), _kValue_, *true*).</del>
+              1. <del>Set _k_ to _k_ + 1.</del>
+              1. <del>Set _n_ to _n_ + 1.</del>
+          1. <del>Else,</del>
+          1. Let _srcBuffer_ be _O_.[[ViewedArrayBuffer]].
+          1. Let _targetBuffer_ be _A_.[[ViewedArrayBuffer]].
+          1. Let _elementSize_ be the Element Size value specified in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for Element Type _srcType_.
+          1. NOTE: If _srcType_ and _targetType_ are the same, the transfer must be performed in a manner that preserves the bit-level encoding of the source data.
+          1. Let _srcByteOffset_ be _O_.[[ByteOffset]].
+          1. Let <del>_targetByteIndex_</del><ins>_targetByteOffset_</ins> be _A_.[[ByteOffset]].
+          1. <del>Let _limit_ be _targetByteIndex_ + _count_ &times; _elementSize_.</del>
+            1. <del>Repeat, while _targetByteIndex_ &lt; _limit_,</del>
+              1. <del>Let _value_ be GetValueFromBuffer(_srcBuffer_, _srcByteIndex_, ~Uint8~, *true*, ~Unordered~).</del>
+              1. <del>Perform SetValueInBuffer(_targetBuffer_, _targetByteIndex_, ~Uint8~, _value_, *true*, ~Unordered~).</del>
+              1. <del>Set _srcByteIndex_ to _srcByteIndex_ + 1.</del>
+              1. <del>Set _targetByteIndex_ to _targetByteIndex_ + 1.</del>
+          1. <ins>Let _n_ be 0.</ins>
+          1. <ins>Repeat, while _k_ &lt; _final_,</ins>
+            1. <ins>Let _srcByteIndex_ be (_k_ &times; _elementSize_) + _srcByteOffset_.</ins>
+            1. <ins>If _srcType_ is the same as _targetType_ and _srcByteIndex_ + _elementSize_ &lt; _srcBufferByteLength_, then</ins>
```

Comment:
Shouldn't we have <= instead of < here?

E.g., srcByteIndex 4, elementSize 1, srcBufferByteLength 5 should still be fine to use the GetValueFromBuffer / SetValueInBuffer below.

Reply:
Yep, exactly right. Fixed.